### PR TITLE
Option dialog configuration fix

### PR DESCRIPTION
**Pull request by CalebKL**
## Description

This PR fixes optionDialog configuration bug by saving and retrieving the data as a Parcelable object. 
Fixes # 
https://github.com/Automattic/pocket-casts-android/issues/1220

## Testing Instructions
1. Go to Profile
2. Go to settings(Top Right)
3. Tap Auto add to up next
4. Click Auto add limit to trigger the option Dialog
5. Change Screen Orientation
6. Observe the dialog doesn't crash

## Screenshots or Screencast 
https://github.com/Automattic/pocket-casts-android/assets/95022986/c6396a2f-88bd-4a5c-a5e1-dc5fc3c464e2

## Checklist
- [ ] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [X] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [ ] I have considered whether it makes sense to add tests for my changes
- [ ] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [ ] Any jetpack compose components I added or changed are covered by compose previews
 
#### I have tested any UI changes...

- [ ] with different themes
- [ ] with a landscape orientation
- [ ] with the device set to have a large display and font size
- [ ] for accessibility with TalkBack

### CODE REVIEW

File: modules/services/views/src/main/java/au/com/shiftyjelly/pocketcasts/views/dialog/OptionsDialog.kt [hunk 1]
```diff
@@ -37,6 +39,32 @@ class OptionsDialog : BottomSheetDialogFragment() {
...
+        private const val KEY_TITLE = "title"
+        private const val KEY_OPTIONS = "options"
+        private const val KEY_ICON_COLOR = "iconColor"
+    }
+
+    @RequiresApi(Build.VERSION_CODES.TIRAMISU)
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        savedInstanceState?.let {
+            options.addAll(it.getParcelableArrayList(KEY_OPTIONS, OptionsDialogOption::class.java) ?: mutableListOf())
```
Comment (mchowning) [inline modules/services/views/src/main/java/au/com/shiftyjelly/pocketcasts/views/dialog/OptionsDialog.kt:53]: This works fine on phones running Android 13, but any phones running older versions of Android will crash when they try to call the `getParcelableArrayList` method that only exists on Android 13. We need to do a version check and call the appropriate method for both Android 13 and for earlier versions. I would suggest creating an extension method similar to what we did with [getSerializableCompat](https://github.com/Automattic/pocket-casts-android/blob/aa9ae49b05e94fae69cdf7c1aa71241a8e0120f9/modules/services/utils/src/main/java/au/com/shiftyjelly/pocketcasts/utils/extensions/Bundle.kt#L1-L0).

**Review by mchowning — CHANGES_REQUESTED** [review]
Thanks for noticing this issue and putting together this fix!

This generally looks good, I just think there are two issues that we need to address before we can merge it.

First, after rotation the device, it doesn't seem like tapping on the buttons is working for me.

<details><summary>Video</summary>



https://github.com/Automattic/pocket-casts-android/assets/4656348/07ccfa4b-1abd-4d80-b86d-de8cde00084e


</details>

Second, we need to make sure this code runs on older versions of Android (this is what I brought up in my other comment).

**Comment by MiSikora** [comment]
@CalebKL Do you intend to work on this or should I close the PR?

**Comment by MiSikora** [comment]
@CalebKL I'm going to close this issue. If you want to follow up on it with the changes please make a new PR.

**Pull request closed without merging**